ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/StatementExecutorTest.java

```diff
@@ -449,6 +451,68 @@ public void shouldEnforceReferentialIntegrity() {
 
   }
 
+  @Test
+  public void shouldFailCreateAsSelectIfExceedActivePersistentQueriesLimit() {
+    // Given:
+    final KsqlConfig ksqlConfig = new KsqlConfig(
+        Collections.singletonMap(
+            KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG, 2));
+    // Create streams and start two persistent queries
+    createStreamsAndTables();
+    // Prepare to try adding a third
+    final Command csasCommand = new Command(
+        "CREATE STREAM user2pv AS "
+            + "select * from pageview"
+            + " WHERE userid = 'user2';",
+        Collections.emptyMap(),
+        ksqlConfig.getAllConfigPropsWithSecretsObfuscated());
+    final CommandId csasCommandId =  new CommandId(CommandId.Type.STREAM,
+        "_CSASGen2",
+        CommandId.Action.CREATE);
+
+    // When:
+    statementExecutor.handleStatement(csasCommand, csasCommandId, Optional.empty());
```

Comment:
you could also pass in a `QueuedCommandStatus` in the third param to allow you to get the status without querying the statementExecutor for it. (I think).   May be better, not sure...

Reply:
OK.  Didn't investigate the option very far. Twas just a suggestion.